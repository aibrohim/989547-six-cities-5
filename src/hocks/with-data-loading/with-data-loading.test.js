import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";

import {withDataLoading} from "./with-data-loading";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withDataLoading(MockComponent);

it(`withDataLoading is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped>
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
