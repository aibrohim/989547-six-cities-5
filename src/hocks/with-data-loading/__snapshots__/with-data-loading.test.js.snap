// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withDataLoading is rendered correctly 1`] = `
<p
  className="status"
>
  Loading
</p>
`;
